eox_tagging/test/test_models.py

```diff
@@ -0,0 +1,183 @@
+"""
+Test classes for Tags model
+"""
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.test import TestCase, override_settings
+
+from eox_tagging.models import Tag
+from eox_tagging.test_utils import CourseFakeModel, EnrollmentsFakeModel
+
+
+@override_settings(EOX_TAGGING_DEFINITIONS=[
+    {"field_name": "tag_value", "validations": ["definition"], "allowed": ["testValue"]},
+    {"field_name": "tag_type", "validations": ["definition"], "allowed": ["testType"]},
+    {"field_name": "tagged_object_name", "validations": ["definition"], "allowed": ["User", "CourseFakeModel"]}])
+@CourseFakeModel.fake_me
+@EnrollmentsFakeModel.fake_me
+class TestTag(TestCase):
+    """Class for testing the Tag model."""
+
+    def setUp(self):
+        """ Model setup used to create objects used in tests."""
+        self.tagged_object = User.objects.create(username="Tag")
+        self.belongs_to_object = User.objects.create(username="User")
+        self.fake_belongs_to_object = CourseFakeModel.objects.create()  # pylint: disable=no-member
+        self.fake_object_tagged = CourseFakeModel.objects.create()  # pylint: disable=no-member
+        self.fake_object_tagged_fail = EnrollmentsFakeModel.objects.create()  # pylint: disable=no-member
+
+        self.test_tag = Tag.objects.create(
+            tag_value="testValue",
+            tag_type="testType",
+            tagged_object=self.tagged_object,
+            belongs_to=self.belongs_to_object,
+        )
+
+        Tag.objects.create(
+            tag_value="testValue",
+            tag_type="testType",
+            tagged_object=self.fake_object_tagged,
+            belongs_to=self.fake_belongs_to_object,
+        )
+
+    @override_settings(EOX_TAGGING_DEFINITIONS=[])
+    def test_empty_setting(self):
+        """ Used to test saving without validations defined"""
+        Tag.objects.create(
```

Comment:
The purpose of the test is create successfully the object without any validation error because there is not any definition set in EOX_TAGGING_DEFINITIONS. That means, that we can create any object without restrictions.
Do you think we can add more to the test?